crates/vm/levm/src/vm.rs

```diff
@@ -610,37 +615,51 @@ impl VM {
     }
 
     /// Gets the max blob gas cost for a transaction that a user is willing to pay.
-    fn get_max_blob_gas_cost(&self) -> Result<U256, VMError> {
-        let blob_gas_used = U256::from(self.env.tx_blob_hashes.len())
+    fn get_max_blob_gas_price(&self) -> Result<U256, VMError> {
+        let blob_gas_used: u64 = self
+            .env
+            .tx_blob_hashes
+            .len()
+            .try_into()
+            .map_err(|_| VMError::Internal(InternalError::ConversionError))?;
+
+        let blob_gas_used: u64 = blob_gas_used
             .checked_mul(BLOB_GAS_PER_BLOB)
             .unwrap_or_default();
 
         let max_blob_gas_cost = self
             .env
             .tx_max_fee_per_blob_gas
             .unwrap_or_default()
-            .checked_mul(blob_gas_used)
+            .checked_mul(blob_gas_used.into())
             .ok_or(InternalError::UndefinedState(1))?;
 
         Ok(max_blob_gas_cost)
     }
 
     /// Gets the actual blob gas cost.
-    fn get_blob_gas_cost(&self) -> Result<U256, VMError> {
-        let blob_gas_used = U256::from(self.env.tx_blob_hashes.len())
+    fn get_blob_gas_price(&self) -> Result<U256, VMError> {
+        let blob_gas_used: u64 = self
+            .env
+            .tx_blob_hashes
+            .len()
+            .try_into()
+            .map_err(|_| VMError::Internal(InternalError::ConversionError))?;
+
+        let blob_gas_price: u64 = blob_gas_used
             .checked_mul(BLOB_GAS_PER_BLOB)
             .unwrap_or_default();
 
         let base_fee_per_blob_gas = self.get_base_fee_per_blob_gas()?;
 
-        let blob_fee = blob_gas_used
+        let blob_fee = blob_gas_price
             .checked_mul(base_fee_per_blob_gas)
-            .ok_or(InternalError::UndefinedState(1))?;
+            .ok_or(VMError::Internal(InternalError::UndefinedState(1)))?;
 
-        Ok(blob_fee)
+        Ok(blob_fee.into())
     }
 
-    pub fn get_base_fee_per_blob_gas(&self) -> Result<U256, VMError> {
+    pub fn get_base_fee_per_blob_gas(&self) -> Result<u64, VMError> {
```

Comment:
this sounds like it should be an U256 because it is a fee. I am not 100% sure but I think it should because in the validations you are comparing it against `tx_max_fee_per_blob_gas`

Reply:
Hello Jere, this was changed here: https://github.com/lambdaclass/ethrex/pull/1528/commits/96b9bf52d81d92cbc91ebccd277e7eb33b2a3375

I added an additional let binding  (`let blob_gas_price: U256 = blob_gas_price.into();`, https://github.com/lambdaclass/ethrex/pull/1528/commits/96b9bf52d81d92cbc91ebccd277e7eb33b2a3375#diff-e20d30a6efb08393eee586d44f3bc4a8df6cbd53ec8ea5bb4d5def88c9487de1R655) simply out of necessity. It was the cleanest way I could think of to cast the value to U256.